### PR TITLE
fix panic when creating consumer with ReceiverQueueSize set to -1

### PR DESCRIPTION
**Pull request by shohi**
Fixes https://github.com/apache/pulsar-client-go/issues/288

### CODE REVIEW

**Comment by wolfstudy** [comment]
> 	// Set to -1 to disable prefetching in consumer
	ReceiverQueueSize int

@shohi Can you remove the comments from `ReceiverQueueSize`?

**Comment by shohi** [comment]
> > // Set to -1 to disable prefetching in consumer
> > ReceiverQueueSize int
> 
> @shohi Can you remove the comments from `ReceiverQueueSize`?

Or change the description?

File: pulsar/consumer_impl.go [hunk 1]
```diff
@@ -74,6 +74,11 @@ func newConsumer(client *client, options ConsumerOptions) (Consumer, error) {
 		options.ReceiverQueueSize = 1000
 	}
 
+	// disable receiver queue if queue size is negative
+	if options.ReceiverQueueSize < 0 {
+		options.ReceiverQueueSize = 0
+	}
```
Comment (wolfstudy) [inline pulsar/consumer_impl.go:80]: Maybe we can fix it using follows way:

```
	if options.ReceiverQueueSize <= 0 {
		options.ReceiverQueueSize = 1000
	}
```

**Comment by wolfstudy** [comment]
> > > // Set to -1 to disable prefetching in consumer
> > > ReceiverQueueSize int
> > 
> > 
> > @shohi Can you remove the comments from `ReceiverQueueSize`?
> 
> Or change the description?

Yes.

File: pulsar/consumer_impl.go (hunk 1 above)
Comment (shohi) [reply pulsar/consumer_impl.go:80]: unbuffered chan may be useful for some cases.

Comment (shohi) [reply pulsar/consumer_impl.go:80]: not sure whether it's a good idea to remove this option. If negative value is totally disallowed (should be reflected in comments), i think it's ok to panic though.

Comment (wolfstudy) [reply pulsar/consumer_impl.go:80]: This is related to the flow mechanism of messages in pulsar. By default, the value of the receiveQueueSize is 1000

Comment (shohi) [reply pulsar/consumer_impl.go:80]: Ok, i'll update the modification and comments according to your suggestions.

**Pull request merged**